### ISSUE TITLE
Hand writing recognition for tamil

### ISSUE THREAD

**Issue by tshrinivasan** [KaniyamFoundation/ProjectIdeas#22]
Hand writing recognition for malayalam
http://gitlab.com/santhoshtr/hand 
https://thottingal.in/projects/hand/

make it for tamil.

Having ligature data in a file for example Tamil.json list like Malayalam.json. The data is very simple . it is x,y coordinates of a sample drawing of a letter. Just draw something on the pad, copy the console output to that json, rebuild. Next time onwards, those strokes will be recognized. This training step is documented in gitlab. This can be done with anybody who knows basics of programming and tamil script.

**Comment by tshrinivasan** [comment]
https://j15h.nu/downloads/handmobile.apk

mobile app for malayalam.
asked for source.

**Comment by Kondasamy** [comment]
I will give a try on this project.

**Comment by tshrinivasan** [comment]
Thanks,

Open it in firefox, open its "Inspect Element", open console.
draw some tamil letter and get its coordinates.

collect coordinates for all tamil letters.

compare with malayalam.json for sample.

**Comment by Kondasamy** [comment]
Please refer the project progress here - https://github.com/Kondasamy/tamil-handwriting-recognition

## Goal:
1. Add Tamil font coordinates for each ligatures and check the correctness
2. Improve the user interface of the website (Dangling UI in Mobile)
3. Improvise the algorithm to accurately detect characters (reduce false positives as much as possible)

**Comment by tshrinivasan** [comment]
from @santhoshtr 

I made some changes to code so that you can chose a language from UI. Added a small dummy training data for Tamil as well. You can see it at https://thottingal.in/projects/hand/
But if you try to add data, don't add lot of letters at once. Add a dozen and stop. This is because I am atill experimenting to see what level of data points granularity required for good performance and good accuracy. That is, I can define a circle using 4 points, 10 points, or 100 points. But this choice has impact on performance and accuracy.
Once we baseline this issue, we can go ahead for wider letter coverage.

**Comment by santhoshtr** [comment]
Thanks. You may not need a "port" of application, since it is mainly data driven. See how I integrated Tamil to application now. Feel free to fork and give pull requests, I can also give access to the repository.
But keep in mind about the data preparation. Go slow for few more days, while I experiment to see some optimal ways to define curve data.
When Jishnu tried to create android app, we realized that the main thread matching algorithm blocks the drawing UI as the data size increases. We might need to go with web worker for the matching part to resolve the issue.

Any ideas for improving is always welcome!

**Comment by Kondasamy** [comment]
Hello Santhosh,

Understand the concerns here. As per your suggestion, will decelerate my
pace on this. Also, I will keep a tab on your project and see the changes.
Please keep this thread posted on any updates. Thanks a lot for the
great support.

Thanks,
Samy
